focuspoints.lrdevplugin/OlympusDelegates.lua

```diff
@@ -44,35 +44,47 @@ OlympusDelegates = {}
 --[[
 -- metaData - the metadata as read by exiftool
 --]]
-function OlympusDelegates.getOlympusAfPoints(photo, metaData)
-    local cameraModel = photo:getFormattedMetadata("cameraModel")
-    local focusPoint = ExifUtils.findValue(metaData, "AF Point Selected")
-    
-  if focusPoint then
-    local focusX, focusY = string.match(focusPoint, "%((%d+)%%,(%d+)")
-    if focusX == nil or focusY == nil then
-        LrErrors.throwUserError("Focus point not found in 'AF Point Selected' metadata tag")
-        return nil, nil
-    end
-    log ("Focus %: " .. focusX .. "," ..  focusY .. "," .. focusPoint  )
-  
-    local dimens = photo:getFormattedMetadata("dimensions")
-    orgPhotoW, orgPhotoH = parseDimens(dimens)
-    log ( "orgPhotoW: " .. orgPhotoW .. " orgPhotoH: ".. orgPhotoH )
-    if orgPhotoW == nil or orgPhotoH == nil then
-        LrErrors.throwUserError("Metadata has no Dimensions")
-        return nil, nil
-    end
-  
-    log ( "Focus px: " .. tonumber(orgPhotoW) * tonumber(focusX)/100 .. "," .. tonumber(orgPhotoH) * tonumber(focusY)/100)
- 
-    if orgPhotoW >= orgPhotoH then
-        return  tonumber(orgPhotoW) * tonumber(focusX)/100, tonumber(orgPhotoH) * tonumber(focusY)/100
-    else
-        return  tonumber(orgPhotoW) * tonumber(focusY)/100, tonumber(orgPhotoH) * tonumber(focusX)/100
-    end
-  else
+function OlympusDelegates.getAfPoints(photo, metaData)
+  local focusPoint = ExifUtils.findFirstMatchingValue(metaData, { "AF Point Selected" })
+  if focusPoint == nil then
     LrErrors.throwUserError("Unsupported Olympus Camera or 'AF Point Selected' metadata tag not found.")
-    return nil, nil    
+    return nil
   end
+
+  local focusX, focusY = string.match(focusPoint, "%((%d+)%%,(%d+)")
+  if focusX == nil or focusY == nil then
+      LrErrors.throwUserError("Focus point not found in 'AF Point Selected' metadata tag")
+      return nil
```

Comment:
execution will stop. nearly positive no return value is needed as it'll never get called.